packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -287,559 +295,688 @@ void main() {
-          isEmpty: true,
-          decoration: const InputDecoration(
-            labelText: labelText,
-            enabledBorder: OutlineInputBorder(
-              borderSide: BorderSide(width: 4),
+      group('when field is in error', () {
+        testWidgets('container has correct height and shape', (WidgetTester tester) async {
+          await tester.pumpWidget(
+            buildInputDecorator(
+              decoration: const InputDecoration(
+                filled: true,
+                labelText: labelText,
+                errorText: errorText,
+              ),
             ),
-            floatingLabelBehavior: FloatingLabelBehavior.always,
-          ),
-        ),
-      );
+          );
 
-      await tester.pump(kTransitionDuration);
+          expect(getContainerRect(tester).height, containerHeight);
+          expect(getBorder(tester), isA<UnderlineInputBorder>());
+          expect(getBorderRadius(tester), const BorderRadius.only(
+            topLeft: Radius.circular(4.0),
+            topRight: Radius.circular(4.0),
+          ));
+        });
 
-      // floatingLabelHeight = 12 (font size 16dps * 0.75 = 12)
-      // labelY = -floatingLabelHeight/2 + borderWidth/2
-      expect(getLabelRect(tester).top, -4.0);
-    });
-  });
+        testWidgets('container has correct color', (WidgetTester tester) async {
+          await tester.pumpWidget(
+            buildInputDecorator(
+              decoration: const InputDecoration(
+                filled: true,
+                labelText: labelText,
+                errorText: errorText,
+              ),
+            ),
+          );
 
-  group('Material3 - InputDecoration label layout', () {
-    testWidgets('The label appears above input', (WidgetTester tester) async {
-      await tester.pumpWidget(
-        buildInputDecorator(
-          decoration: const InputDecoration(
-            label: customLabel,
-          ),
-        ),
-      );
+          final ThemeData theme = Theme.of(tester.element(findDecorator()));
+          expect(findBorderPainter(), paints
+            ..path(
+              style: PaintingStyle.fill,
+              color: theme.colorScheme.surfaceContainerHighest,
+            )
+          );
+        });
 
-      // Overall height for this InputDecorator is 56dp on mobile:
-      //    8 - top padding
-      //   12 - floating label (font size = 16 * 0.75, line height is forced to 1.0)
-      //    4 - gap between label and input (this is not part of the M3 spec)
-      //   24 - input text (font size = 16, line height = 1.5)
-      //    8 - bottom padding
-      expect(getDecoratorRect(tester).size, const Size(800.0, 56.0));
-      expect(getCustomLabelRect(tester).top, 8.0);
-      expect(getCustomLabelRect(tester).bottom, 20.0);
-      expect(getInputRect(tester).top, 24.0);
-      expect(getInputRect(tester).bottom, 48.0);
-    });
+        testWidgets('active indicator has correct height and color', (WidgetTester tester) async {
```

Comment:
Here's another test where the title says "height" but you check the weight. It looks like there are others.